### PR TITLE
Hm21

### PR DESCRIPTION
**Pull request by floriangeyer21**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/order/CancelingOrderController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        /* String id = req.getParameter("id");
+        Long userId = Long.valueOf(id);*/
```
Comment (margarita-ignatieva) [inline src/main/java/com/internet/shop/controllers/order/CancelingOrderController.java]: Why do you use comments here?

**Review by margarita-ignatieva — APPROVED** [review]

File: src/main/java/com/internet/shop/controllers/order/CancelingOrderController.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/order/canceling")
+public class CancelingOrderController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/CancelingOrderController.java]: the user can't cancel the order after it is completed
let's remove this controller

File: src/main/java/com/internet/shop/controllers/order/CreateOrderController.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        /* String id = req.getParameter("id");
+        Long userId = Long.valueOf(id);*/
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/CreateOrderController.java]: remove the comments from everywhere

File: src/main/java/com/internet/shop/controllers/order/DeleteOrderForAdminController.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import com.internet.shop.lib.Injector;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/order/delete")
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/DeleteOrderForAdminController.java]:
```suggestion
@WebServlet("/orders/delete")
```

File: src/main/java/com/internet/shop/controllers/order/DeleteOrderForAdminController.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+public class DeleteOrderForAdminController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/DeleteOrderForAdminController.java]:
```suggestion
public class DeleteOrderController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/order/DeleteOrderForAdminController.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        /* String id = req.getParameter("id");
+        Long userId = Long.valueOf(id);*/
+        orderService.delete(USER_ID);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/DeleteOrderForAdminController.java]: why USER_ID and not orderId?

File: src/main/java/com/internet/shop/controllers/order/GetAllOrdersForAdminController.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/order/all-for-admin")
+public class GetAllOrdersForAdminController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/GetAllOrdersForAdminController.java]:
```suggestion
public class GetAllOrdersController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/order/GetOrdersDetailController.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/order/getDetail")
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/GetOrdersDetailController.java]:
```suggestion
@WebServlet("/orders/info")
```

File: src/main/java/com/internet/shop/controllers/order/GetOrdersDetailController.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+public class GetOrdersDetailController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/GetOrdersDetailController.java]:
```suggestion
public class GetOrderDetailsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/order/GetOrdersDetailController.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        // String id = req.getParameter("id");
+        // Long ordersId = Long.valueOf(id);
+        Order order = orderService.getById(USER_ID);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/GetOrdersDetailController.java]: same here - why USER_ID?

File: src/main/java/com/internet/shop/controllers/order/GetOrdersForUserController.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/order/all")
+public class GetOrdersForUserController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/GetOrdersForUserController.java]:
```suggestion
@WebServlet("/user/orders")
public class GetAllUserOrdersController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/product/GetAllProductsForAdminController.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/product/all-for-admin")
+public class GetAllProductsForAdminController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/product/GetAllProductsForAdminController.java]:
```suggestion
@WebServlet("/products/manage")
public class ManageProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/product/GetAllProductsForUsersController.java [hunk 13]
```diff
@@ -12,7 +12,7 @@
 import javax.servlet.http.HttpServletResponse;
 
 @WebServlet("/product/all")
-public class GetAllProductsController extends HttpServlet {
+public class GetAllProductsForUsersController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/product/GetAllProductsForUsersController.java]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/user/DeleteUsersController.java [hunk 14]
```diff
@@ -24,8 +23,8 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         String userId = req.getParameter("id");
         Long id = Long.valueOf(userId);
-        ShoppingCart shoppingCart = shoppingCartService.getByUserId(id);
-        shoppingCartService.delete(shoppingCart.getId());
+        /* ShoppingCart shoppingCart = shoppingCartService.getByUserId(id);
+        shoppingCartService.delete(shoppingCart.getId());*/
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/user/DeleteUsersController.java]: why have you commented this?
your shopping cart should be deleted with the user

File: src/main/java/com/internet/shop/model/Order.java [hunk 15]
```diff
@@ -6,12 +6,17 @@
     private Long id;
     private List<Product> products;
     private Long userId;
+    private Double price;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/Order.java]: please remove this field
we should not store this in Order - we will have trouble when we will connect to the DB

File: src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java [hunk 16]
```diff
@@ -19,6 +20,11 @@
     @Override
     public Order completeOrder(ShoppingCart shoppingCart) {
         Order order = new Order(shoppingCart.getUserId(), List.copyOf(shoppingCart.getProducts()));
+        Double price = (Double) List.copyOf(shoppingCart.getProducts()).stream()
+                .map(Product::getPrice)
+                .reduce(Double::sum)
+                .get();
+        order.setPrice(price);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/service/impl/OrderServiceImpl.java]: let's remove this

File: src/main/webapp/WEB-INF/views/order/allOrdersForAdmin.jsp [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+        }
+    </style>
+</head>
+<body class="body1">
+<h1 class="font">User orders</h1>
+<table class="table_dark">
+    <tr>
+        <th>ID</th>
+        <th>User ID</th>
+        <th>Price</th>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/allOrdersForAdmin.jsp]: let's remove `Price` column

File: src/main/webapp/WEB-INF/views/order/allOrdersForAdmin.jsp [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+            </td>
+        </tr>
+    </c:forEach>
+</table>
+
+<div class="font : font1">
+    <a href="${pageContext.request.contextPath}/">To main page</a>
+</div>
+</body>
+</html>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/allOrdersForAdmin.jsp]: add a blank line to the end

File: src/main/webapp/WEB-INF/views/order/allUserOrders.jsp [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+    </style>
+</head>
+<body class="body1">
+<h1 class="font">User orders</h1>
+<table class="table_dark">
+    <tr>
+        <th>ID</th>
+        <th>Price</th>
+        <th>Detail</th>
+        <th>Canceling</th>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/allUserOrders.jsp]: let's remove `price` and `canceling` columns

File: src/main/webapp/WEB-INF/views/product/all-for-admin.jsp [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+</table>
+<h3><p></p></h3>
+<form action="${pageContext.request.contextPath}/product/add">
+    <button type="submit" class="test">Add product</button>
+</form>
+<div class="font : font1">
+    <a href="${pageContext.request.contextPath}/">To main page</a>
+</div>
+</body>
+</html>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/product/all-for-admin.jsp]: empty line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/order/CreateOrderController.java [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+import com.internet.shop.service.OrderService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/order/create")
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/CreateOrderController.java]:
```suggestion
@WebServlet("/orders/create")
```

File: src/main/java/com/internet/shop/controllers/order/DeleteOrderController.java [hunk 22]
```diff
@@ -0,0 +1,27 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String id = req.getParameter("id");
+        Long userId = Long.valueOf(id);
+        orderService.delete(ORDER_ID);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/DeleteOrderController.java]: can you explain please why are you receiving userId as request parameter
where do you use it and why is orderId hardcoded?

File: src/main/java/com/internet/shop/controllers/order/GetAllOrdersController.java [hunk 23]
```diff
@@ -0,0 +1,27 @@
...
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/order/all-for-admin")
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/GetAllOrdersController.java]:
```suggestion
@WebServlet("/orders")
```

File: src/main/java/com/internet/shop/controllers/order/GetOrderDetailsController.java [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+public class GetOrderDetailsController extends HttpServlet {
+    private static final Long ORDER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Order order = orderService.getById(ORDER_ID);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/order/GetOrderDetailsController.java]: let's receive orderId as a request parameter here

File: src/main/java/com/internet/shop/controllers/product/AddProductController.java [hunk 25]
```diff
@@ -30,6 +30,6 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
         Double price = Double.valueOf(inputPrice);
         Product product = new Product(name, price);
         productService.create(product);
-        resp.sendRedirect(req.getContextPath() + "/");
+        resp.sendRedirect(req.getContextPath() + "/product/all-for-admin");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/product/AddProductController.java]:
```suggestion
        resp.sendRedirect(req.getContextPath() + "/products/manage");
```

File: src/main/java/com/internet/shop/controllers/product/DeleteProductController.java [hunk 26]
```diff
@@ -21,6 +21,6 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         String userId = req.getParameter("id");
         Long id = Long.valueOf(userId);
         productService.delete(id);
-        resp.sendRedirect(req.getContextPath() + "/product/all");
+        resp.sendRedirect(req.getContextPath() + "/product/all-for-admin");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/product/DeleteProductController.java]:
```suggestion
        resp.sendRedirect(req.getContextPath() + "/products/manage");
```
after changing the mapping be sure that you updated it everywhere you used it

File: src/main/webapp/WEB-INF/views/order/allOrdersForAdmin.jsp [hunk 27]
```diff
@@ -0,0 +1,60 @@
...
+    <c:forEach var="order" items="${orders}">
+        <tr>
+            <td>
+                <c:out value="${order.id}"/>
+            </td>
+            <td>
+                <c:out value="${order.userId}"/>
+            </td>
+            <td>
+                <a href="${pageContext.request.contextPath}/orders/info">Detail</a>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/allOrdersForAdmin.jsp]: add a param here

File: src/main/webapp/WEB-INF/views/order/allUserOrders.jsp [hunk 28]
```diff
@@ -0,0 +1,52 @@
...
+        <th>ID</th>
+        <th>Detail</th>
+    </tr>
+    <c:forEach var="order" items="${orders}">
+        <tr>
+            <td>
+                <c:out value="${order.id}"/>
+            </td>
+            <td>
+                <a href="${pageContext.request.contextPath}/orders/info">Detail</a>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/allUserOrders.jsp]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**